packages/flutter/lib/src/material/theme_data.dart

```diff
@@ -1073,6 +1078,27 @@ class ThemeData with Diagnosticable {
   /// {@endtemplate}
   final VisualDensity visualDensity;
 
+  /// A temporary flag used to opt-in to new Material 3 features.
+  ///
+  /// If true, then components that have been migrated to Material 3 will
+  /// start using new colors, typography and other features of Material 3.
+  /// If false, they will use the Material 2 look and feel.
+  ///
+  /// Currently no components have been migrated to support Material 3.
+  /// As they are updated to include Material 3 support this documentation
+  /// will be modified to indicate exactly what widgets this flag will affect.
+  ///
+  /// [useMaterial3] defaults to false. After all the migrated components
```

Comment:
Can we add a disclaimer that if you enable this flag now, your app could break visually often during the next couple months?

Reply:
Yeah, I wasn't sure how to word smith that. I will take a stab at it though. Thx.